File: src/python/osc-test.py
# -*- coding: utf-8 -*-
"""
Created on Thu Aug 13 11:44:13 2020

@author: gonca
"""

import rigstim
from datetime import datetime
from rigstim import WallType as Wall
from pyOSC3fix.OSC3 import OSCStreamingClient
import threading
import time

def default_handler(address, *args):
    print(f"DEFAULT {address}: {args}")

ip = "127.0.0.1"
request_port = 4002

sessiontime = datetime.now()
metadata = "{0}_{1}-{2}-{3}_S1".format(
        sessiontime.date(),
        sessiontime.hour,
        sessiontime.minute,
        sessiontime.second)
client = OSCStreamingClient()  # Create client
try:
    client.connect((ip, request_port))
    rig = rigstim.RigClient(client)
    rig.dataset('Data')

    rig.resource("Videos/Blink")
    rig.preload()
    
    rig.experiment(metadata)
    rig.gratings(size=30, x=-15, y=-5, angle=0, freq=0.1, duration=2.0) # grating 1
    rig.gratings(size=15, x=15, y=-5, angle=45, freq=0.1, duration=2.0, speed=1) # grating 2
    rig.video("Blink", y=20, speed=2.0, onset=1.0, duration=2.0) # video 1
    rig.start()

    rig.receive()  # Wait for end trial
    rig.clear()
    
    rig.experiment(metadata)
    rig.gratings(size=120, angle=30, freq=0.1, duration=2.0) # go gratings
    rig.go(suppress=1000, start=500, duration=1000.0, threshold=2) # go trial
    rig.receive()  # Wait for end trial

    rig.experiment(metadata)
    rig.gratings(size=120, angle=0, freq=0.1, duration=2.0) # nogo gratings
    rig.nogo(suppress=500, start=0.0, duration=2000.0) # no-go trial
    rig.receive()  # Wait for end trial

    rig.experiment(metadata)
    rig.interaction('rewardLick', [2, 1]) # lickthreshold, max activations
    rig.interaction('endLick', [3, 3000.0]) # lickthreshold, delay
    rig.tile(Wall.LEFT, position=0, extent=1, texture="White")
    rig.tile(Wall.RIGHT, position=0, extent=1, texture="Black")
    
    rig.interaction('teleportLick', [0.0, 3, 3])
    rig.tile(Wall.LEFT, position=1, extent=1, texture="Black")
    
    rig.interaction('rewardEntry', [1000.0, 2])
    rig.tile(Wall.RIGHT, position=1, extent=1, texture="White")
    
    rig.interaction('endEntry', 2000.0)
    rig.tile(Wall.LEFT, position=2, extent=1, texture="White")
    
    rig.interaction('teleportEntry', [0.0, 2])
    rig.tile(Wall.RIGHT, position=2, extent=1, texture="Black")
    
    rig.interaction('gainEntry', [0.1, 1])
    rig.tile(Wall.TOP, position=1, extent=1, texture="Black")
    rig.corridor(length=3.0, width=1.2, height=1.0, x=0.0, y=0.0, position=0.0)
    rig.receive()  # Wait for end trial
    
    rig.experiment(metadata)
    rig.replay(metadata, 3)
    rig.receive()  # Wait for end trial

    rig.experiment(None)
finally:
    client.close()

File: src/matlab/osctest.m
%u = udp('158.109.215.49', 4002, 'LocalPort', 4007);
%u = tcpip('158.109.215.49', 4002,"NetworkRole","client");
osc = OscTcp('158.109.215.49', 4002);

% Mouse Test
% osc = OscTcp('127.0.0.1', 4242);
% while(true)
%     res = osc.receive()
% end



%u= tcpclient('158.109.215.49', 4002);
rig = Rig(osc);

rig.resource("Videos/Blink");
rig.preload();

g1.size = 30;
g1.x = -15;
g1.y = -5;
g1.angle = 0;
g1.freq = 0.1;
g1.duration = 2.0;
rig.gratings(g1); % grating 1

g2.size = 15;
g2.x = 15;
g2.y = -5;
g2.angle = 45;
g2.freq = 0.1;
g2.duration = 2.0;
g2.speed = 1;
rig.gratings(g2); % grating 2

v.name = "Blink";
v.y = 20;
v.speed = 2.0;
v.onset = 1.0;
v.duration = 2.0;
rig.video(v); % video 1
rig.start();
osc.receive()

rig.clear();


gg.size = 120;
gg.angle = 30;
gg.freq = 0.1;
gg.duration = 2.0;
rig.gratings(gg); % go gratings
rig.go(1000, 500, 1000, 2); % go trial
osc.receive() % Wait for end trial

ng.size = 120;
ng.angle = 0;
ng.freq = 0.1;
ng.duration = 2.0;
rig.gratings(ng); % nogo gratings
rig.nogo(500, 500, 1000, 1); % no-go trial
osc.receive() % Wait for end trial

rig.interaction('rewardLick', "ii", {2, 1}); % lickthreshold, max activations
rig.interaction('endLick', "if", {3, 3000.0}); % lickthreshold, delay
t0.wall = Wall.Left;
t0.position = 0;
t0.extent = 1;
t0.texture = "White";
rig.tile(t0);

t1.wall = Wall.Right;
t1.position = 0;
t1.extent = 1;
t1.texture = "Black";
rig.tile(t1);

rig.interaction('teleportLick', "fii", {0.0, 3, 3});
t2.wall = Wall.Left;
t2.position = 1;
t2.extent = 1;
t2.texture = "Black";
rig.tile(t2);

rig.interaction('rewardEntry', "fi", {1000.0, 2});
t3.wall = Wall.Right;
t3.position = 1;
t3.extent = 1;
t3.texture = "White";
rig.tile(t3);

rig.interaction('endEntry', "f", {2000.0});
t4.wall = Wall.Left;
t4.position = 2;
t4.extent = 1;
t4.texture = "White";   
rig.tile(t4);

rig.interaction('teleportEntry', "fi", {0.0, 2});
t5.wall = Wall.Right;
t5.position = 2;
t5.extent = 1;
t5.texture = "Black";
rig.tile(t5);

rig.interaction('gainEntry', "fi", {0.1, 1});
t6.wall = Wall.Top;
t6.position = 1;
t6.extent = 1;
t6.texture = "Black";
rig.tile(t6);

c.length = 3.0;
c.width = 1.2;
c.height = 1.0;
c.x=0.0;
c.y=0.0;
c.position=0.0;
rig.corridor(c);
osc.receive() % Wait for end trial

rig.close();